plugins/setoid_ring/g_newring.mlg

```diff
@@ -133,7 +135,9 @@ VERNAC COMMAND EXTEND AddSetoidField CLASSIFIED AS SIDEFF
 | [ "Print" "Fields" ] => {Vernacextend.classify_as_query} -> {
     Feedback.msg_notice (strbrk "The following field structures have been declared:");
     Spmap.iter (fun fn fi ->
-      let sigma, env = Pfedit.get_current_context () in
+      (* We use the global env here as this shouldn't contain proof
+         data. *)
+      let sigma, env = let e = Global.env () in Evd.from_env e, e in
```

Comment:
wrong env

Reply:
Why is the wrong env?

Reply:
The behavior is different from the one before. Printing does rename stuff is there are collisions with the given context. You are right that the term being printed lives in the global env. I've no clue why they picked the proof env there. In some cases it is intentional, eg `Print` + `apply with x0 := t` IIRC.

Reply:
IMHO it is not obviously wrong to take the proof env, so let's keep things as they were.

Reply:
Here again, let's keep orthogonal changes separated (if easy). If we can access the proof state, let's just `Option.cata` on it.

Reply:
This is still to be fixed.